src/arviz_plots/plots/ppcplot.py

```diff
@@ -0,0 +1,529 @@
+    # datatree provided
+    if observed:
+        for group_name in (predictive_data_group, "observed_data"):
+            if group_name not in dt.children:
+                raise TypeError(f'`data` argument must have the group "{group_name}" for ppcplot')
+    else:
+        if f"{predictive_data_group}" not in dt.children:
+            raise TypeError(f'`data` argument must have the group "{group}_predictive" for ppcplot')
+
+    # making sure kde type is one of these three
+    if kind.lower() not in ("kde", "cumulative", "scatter"):
+        raise TypeError("`kind` argument must be either `kde`, `cumulative`, or `scatter`")
+    if kind == "cumulative":
+        kind = "ecdf"
+
+    # initializaing data_pairs as empty dict in case pp and observed data var names are same
+    if data_pairs is None:
+        data_pairs = {}
+
+    if backend is None:
+        if plot_collection is None:
+            backend = rcParams["plot.backend"]
+        else:
+            backend = plot_collection.backend
+    # getting plot backend to get default colors to put in default kwargs for visual elements
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+
+    # pp distribution group plotting logic
+    pp_distribution = process_group_variables_coords(
+        dt,
+        group=predictive_data_group,
+        var_names=(
+            None
+            if var_names is None
+            else [data_pairs.get(var_name, var_name) for var_name in var_names]
+        ),
+        filter_vars=filter_vars,
+        coords=coords,
+    )
+
+    # creating random number generator
+    if random_seed is not None:
+        rng = np.random.default_rng(random_seed)
+    else:
+        rng = np.random.default_rng()
+
+    # picking random pp dataset sample indexes and subsetting pp_distribution accordingly
+    # total_pp_samples = plot_collection.data.sizes["chain"] * plot_collection.data.sizes["draw"]
+    total_pp_samples = np.prod(
+        [pp_distribution.sizes[dim] for dim in sample_dims if dim in pp_distribution.dims]
+    )
+    if num_pp_samples is None:
+        if kind == "scatter":
+            num_pp_samples = min(5, total_pp_samples)
+        else:
+            num_pp_samples = total_pp_samples
+
+    if (
+        not isinstance(num_pp_samples, Integral)
+        or num_pp_samples < 1
+        or num_pp_samples > total_pp_samples
+    ):
+        raise TypeError(f"`num_pp_samples` must be an integer between 1 and {total_pp_samples}.")
+
+    pp_sample_ix = rng.choice(total_pp_samples, size=num_pp_samples, replace=False)
+
+    # print(f"\npp_sample_ix: {pp_sample_ix!r}")
+
+    # stacking sample dims into a new 'ppc_dim' dimension
+    pp_distribution = pp_distribution.stack(ppc_dim=sample_dims)
+
+    # Select the desired samples
+    pp_distribution = pp_distribution.isel(ppc_dim=pp_sample_ix)
+
+    # renaming sample_dims so that rest of plot will consider this as sample_dims
+    sample_dims = ["ppc_dim"]
```

Comment:
I think we should skip all these steps if num_pp_samples == total_pp_samples. I think nothing within the function itself should change (test probably will) and it would allow using aesthetics for chain/draw when plotting all samples

Reply:
we might also want to distinguish between length 1 sample_dims (in which case it is not necessary to stack nor to rename and overwrite its value) and >1 length sample_dims (in which case stacking is necessary)